### PR TITLE
Fixes #191, crash on .xlsx written by non-Excel tools with non-standard xml attributes

### LINKED ISSUE #191: Reading a certain Excel (XLSX) file crashes R
Reading a certain XSLX file using `read_excel()` consistently crashes R, on both Windows (latest CRAN release, version 0.1.1) and Linux (latest GitHub release as of 2016-07-10). It results in this error message:

```
*** caught segfault ***
address 0x8, cause 'memory not mapped'

Traceback:
1: .Call("readxl_read_xlsx_", PACKAGE = "readxl", path, sheet, col_names,     col_types, na, nskip)
2: read_xlsx_(path, sheet, col_names = col_names, col_types = col_types,     na = na, nskip = skip)
3: read_xlsx(path, sheet, col_names, col_types, na, skip)
4: read_excel("crashing.xlsx")

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace
```

The file can be successfully loaded in both MS Excel and LibreOffice Calc. However, it was not created by Excel, but by some other tool. If I open it in Excel and save it as a new file, `readxl` can successfully open it.

There’s not any confidential data in the file, but I’d rather not make it public. So I’m sending it privately to @hadley.

### PR DESCRIPTION
**Pull request by jennybc**
First fruit of my C++ tutoring from @jimhester.

Prevents attempt to read value of a nonexistent attribute (can occur in xlsx written by something other than Excel).

### CODE REVIEW

File: src/XlsxWorkBook.h [hunk 1]
```diff
@@ -120,23 +126,27 @@ class XlsxWorkBook {
...
+    }
 
     int i = 0;
     for (rapidxml::xml_node<>* cellXf = cellXfs->first_node();
          cellXf; cellXf = cellXf->next_sibling()) {
-      int formatId = atoi(cellXf->first_attribute("numFmtId")->value());
-      if (isDateTime(formatId, customDateFormats))
-        dateStyles_.insert(i);
-      ++i;
+      if (cellXf->first_attribute("numFmtId") != NULL) {
```
Comment (hadley) [inline src/XlsxWorkBook.h]: I think this is easier to read if you skip the bad inputs:

```R
if (cellXf->first_attribute("numFmtId") == NULL)
  next;
```

Comment (jennybc) [inline src/XlsxWorkBook.h]: Done. Unless you want the `else{...}`? And bullet added to NEWS.

**Comment by jennybc** [comment]
The appveyor failure is spurious, I think. It's refusing to attempt the build vs. building and failing. I can't tell why it thinks this PR is un-mergeable.

**Comment by hadley** [comment]
LGTM

**Pull request merged**